src/main/io/gps.c

```diff
@@ -878,30 +1364,40 @@ void gpsUpdate(timeUs_t currentTimeUs)
 
         case GPS_STATE_LOST_COMMUNICATION:
             gpsData.timeouts++;
-            if (gpsConfig()->autoBaud) {
+            // commented out, as it is dangerous and should not be used, fix for a correct auto baud in another pr
+            /*if (gpsConfig()->autoBaud) {
                 // try another rate
                 gpsData.baudrateIndex++;
                 gpsData.baudrateIndex %= GPS_INIT_ENTRIES;
-            }
+            }*/
             gpsSol.numSat = 0;
             DISABLE_STATE(GPS_FIX);
             gpsSetState(GPS_STATE_INITIALIZING);
             break;
 
         case GPS_STATE_RECEIVING_DATA:
             // check for no data/gps timeout/cable disconnection etc
-            if (millis() - gpsData.lastMessage > GPS_TIMEOUT) {
+            DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 2, millis() - gpsData.lastMessage);
+            if (millis() - gpsData.lastMessage > GPS_TIMEOUT_MS(gpsData.updateRate)) {
+                DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 3, debug[3] + 1);
                 gpsSetState(GPS_STATE_LOST_COMMUNICATION);
+                gpsPackageCounter = 0;
+                DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 1, gpsPackageCounter);
 #ifdef USE_GPS_UBLOX
             } else {
                 if (gpsConfig()->autoConfig == GPS_AUTOCONFIG_ON) { // Only if autoconfig is enabled
                     switch (gpsData.state_position) {
+                        case UBLOX_DETECT_UNIT:
+                            gpsData.state_position = UBLOX_INITIALIZE;
+                            break;
                         case UBLOX_INITIALIZE:
                             if (!isConfiguratorConnected()) {
-                                if (gpsData.ubloxUseSAT) {
-                                    ubloxSetMessageRate(CLASS_NAV, MSG_SAT, 0); // disable SAT MSG
+                                if (ubloxHasValSetGet()) {
```

Comment:
This should be an internal problem to `ubloxSetMessageRate`, the `if-else` should not be here.

Reply:
has been fixed in upcoming PR

Reply:
Should be fixed now